src/data/services/DiscoveryDataApiService.test.js

```diff
@@ -20,3 +31,32 @@ describe('fetchOrganizationUsers', () => {
     });
   });
 });
+
+describe('fetchBulkOperationTask', () => {
+  const taskId = 1;
+  const expectedUrl = `${process.env.DISCOVERY_API_BASE_URL}/api/v1/bulk_operation_tasks/${taskId}/`;
+  const get = jest.spyOn(axios, 'get');
+
+  it('should fetch task successfully', async () => {
+    const mockData = mockTask;
+    get.mockResolvedValue({ data: mockData });
+
+    const result = await DiscoveryDataApiService.fetchBulkOperationTask(taskId);
+    expect(get).toHaveBeenCalledWith(expectedUrl);
+    expect(result.data).toEqual(mockData);
+  });
+
+  it('should return null on 404 response', async () => {
+    get.mockRejectedValue({ response: { status: 404 } });
+
+    const result = await DiscoveryDataApiService.fetchBulkOperationTask(taskId).catch(e => e);
```

Comment:
Do we usually use catch with await? I feel like we can use `.rejects` and expect a value in case a promise is rejected. Something like

`await expect(promise).rejects.toEqual(expectedError);`

Reply:
Actually, I think both are fine. It's about preference of what you'd like. However, I've updated the code as you suggested.